entitydef/user-entitymodel_view.xml

```diff
@@ -0,0 +1,61 @@
+<entitymodel xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+             xsi:noNamespaceSchemaLocation="https://ofbiz.apache.org/dtds/entitymodel.xsd">
+    <!-- ========================================================= -->
+    <!-- ======================== Defaults ======================= -->
+    <!-- ========================================================= -->
+    <title>View Entity of Custom UserMgr application</title>
+    <description>None</description>
+    <copyright></copyright>
+    <version></version>
+    
+    <view-entity entity-name="UserAndParty" package-name="org.apache.ofbiz.user">
+        <member-entity entity-alias="UL" entity-name="UserLogin"/>
+        <member-entity entity-alias="P" entity-name="Person"/>
+        <member-entity entity-alias="PCM" entity-name="PartyContactMech"/>
+        <member-entity entity-alias="PA" entity-name="PostalAddress"/>
+        <member-entity entity-alias="TN" entity-name="TelecomNumber"/>
+        <member-entity entity-alias="CM" entity-name="ContactMech"/>
+        <member-entity entity-alias="PCMP" entity-name="PartyContactMechPurpose"/>
+        <member-entity entity-alias="ULSG" entity-name="UserLoginSecurityGroup"/>
+        <alias name="userLoginId" entity-alias="UL" field="userLoginId" prim-key=""/>
+        <alias name="partyId" entity-alias="P" field="partyId"/>
+        <alias name="firstName" entity-alias="P" field="firstName"/>
+        <alias name="middleName" entity-alias="P" field="middleName"/>
+        <alias name="lastName" entity-alias="P" field="lastName"/>
+        <alias name="gender" entity-alias="P" field="gender"/>
+        <alias name="birthDate" entity-alias="P" field="birthDate"/>
+        <alias name="contactMechId" entity-alias="PCM" field="contactMechId"/>
+        <alias name="emailAddress" entity-alias="CM" field="infoString"/>
+        <alias name="address1" entity-alias="PA" field="address1"/>
+        <alias name="address2" entity-alias="PA" field="address2"/>
+        <alias name="city" entity-alias="PA" field="city"/>
+        <alias name="postalCode" entity-alias="PA" field="postalCode"/>
+        <alias name="countryGeoId" entity-alias="PA"/>
+        <alias name="countryCode" entity-alias="TN"/>
+        <alias name="areaCode" entity-alias="TN"/>
+        <alias name="contactNumber" entity-alias="TN"/>
+        <alias name="groupId" entity-alias="ULSG"/>
+        <view-link entity-alias="UL" rel-entity-alias="P" rel-optional="true">
```

Comment:
why we set rel-optional="true" here?

Reply:
The "rel-optional" attribute is used to indicate whether the related entity is optional or not. In this case, "rel-optional='true'" is used because it is possible that certain "userLogin" may not be associated with a "party" but still exists. Therefore, in such situations, we still want to retrieve the records. The "rel-optional='true'" performs a left outer join and we get the records for the "userLogin" details with NULL in the "party" details. Similarly, not all contact mechanisms may be associated with all postal addresses, telecom numbers, or contact mech purposes. That's why we use "rel-optional='true'".